NDTensors/src/Unwrap/test/runtests.jl

```diff
@@ -138,4 +138,31 @@ include("../../../test/device_list.jl")
   x = dev(randn(elt, 4, 4))
   permutedims!(expose(y), expose(x), (2, 1))
   @test NDTensors.cpu(y) == transpose(NDTensors.cpu(x))
+
+  ##########################################
+  ### Testing an issue with CUDA&Metal transpose/adjoint mul
+  A = dev(randn(elt, (3, 2)))
+  B = dev(randn(elt, (3, 4)))
+  C = dev(randn(elt, (4, 2)))
+  Cp = copy(C)
+
+  ## This fails with scalar indexing 
+  #mul!(transpose(C), transpose(A), B, true, false)
+  mul!(C, transpose(B), A, true, false)
+  mul!(expose(transpose(Cp)), expose(transpose(A)), expose(B), true, false)
+  @test C ≈ Cp
+  Cp = zero(C)
+  ## Try calling mul!! with transposes to verify that code works
+  Cpt = NDTensors.mul!!(transpose(Cp), transpose(A), B, true, false)
+  @test transpose(Cpt) ≈ C
+
+  Cp = zero(C)
+  ## This fails with scalar indexing 
+  #mul!(C', A', B, true, false)
```

Comment:
It's good to keep track of examples of failures.

If we disable scalar indexing in the tests then we can mark this as broken to keep track of it with:
```julia
@test_broken mul!(C', A', B, true, false)
```

Reply:
That makes sense in the case they are fixed by `Metal` or `CUDA` we can remove our case specific `Expose` based code. I made this change for both Adjoint and transpose

Reply:
Yeah good point. Our overloads of `Exposed` and these tests will serve as a good record of what needs to be fixed upstream in the GPU packages, once the dust settles on our implementation we should definitely raise issues upstream for the cases we have found.